core/src/main/scala/cats/data/WriterT.scala

```diff
@@ -149,6 +157,15 @@ private[data] sealed trait WriterTMonad[F[_], L] extends WriterTApplicative[F, L
     fa.flatMap(f)
 }
 
+private[data] sealed trait WriterTMonadError[F[_], L, E] extends MonadError[WriterT[F, L, ?], E] with WriterTMonad[F, L] {
+  override implicit def F0: MonadError[F, E]
+
+  def raiseError[A](e: E): WriterT[F, L, A] = WriterT(F0.raiseError[(L, A)](e))
+
+  def handleErrorWith[A](fa: WriterT[F, L, A])(f: E => WriterT[F, L, A]): WriterT[F, L, A] =
+    WriterT(F0.handleErrorWith(fa.run)(e => f(e).run))
```

Comment:
@lukiano sorry I'm just seeing this. I don't think we can do what you have recommended. Let's consider `Future` as `F` (and thus `Throwable` as `E`). If the `Future` fails, then there is no log. We _do_ have a log if the future succeeds, but in that case the function that we pass to `handleErrorWith` will never be called, because there is no error to handle.

Does that sound right?